main.tf

```diff
@@ -234,7 +246,7 @@ resource "aws_s3_bucket" "s3_bucket" {
       tags                                   = lookup(lifecycle_rule.value, "tags", null)
 
       dynamic "expiration" {
-        for_each = lookup(lifecycle_rule.value, "expiration", [])
+        for_each = lookup(lifecycle_rule.value, "expiration", [{}])
```

Comment:
Latest updated due to a bug where later plans would remove a blank\default expiration block without this modification.